javax/ws/rs/core/MediaType.cpp

```diff
@@ -0,0 +1,196 @@
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "MediaType.hpp"
+#include "../../../../java/lang/StringBuilder/StringBuilder.hpp"
+#include "MediaTypeException.hpp"
+
+using namespace Javax::Ws;
+
+#define CHAR_NOT_FOUND -1
+
+String MediaType::APPLICATION_ATOM_XML          = "application/atom+xml";
+String MediaType::APPLICATION_FORM_URLENCODED   = "application/x-www-form-urlencoded";
+String MediaType::APPLICATION_JSON              = "application/json";
+String MediaType::APPLICATION_OCTET_STREAM      = "application/octet-stream";
+String MediaType::APPLICATION_SVG_XML           = "application/svg+xml";
+String MediaType::APPLICATION_XHTML_XML         = "application/xhtml+xml";
+String MediaType::APPLICATION_XML               = "application/xml";
+String MediaType::MEDIA_TYPE_WILDCARD           = "*";
+String MediaType::MULTIPART_FORM_DATA           = "multipart/form-data";
+String MediaType::TEXT_HTML                     = "text/html";
+String MediaType::TEXT_PLAIN                    = "text/plain";
+String MediaType::TEXT_XML                      = "text/xml";
+String MediaType::WILDCARD                      = "*/*";
+
+MediaType::MediaType() {
+    type = MEDIA_TYPE_WILDCARD;
+    subtype = MEDIA_TYPE_WILDCARD;
+}
+
+MediaType::MediaType(const String &type, const String &subtype) {
+    if (type.isEmpty()) {
+        throw MediaTypeException("`type` must not be empty");
+    }
+
+    if (subtype.isEmpty()) {
+        throw MediaTypeException("`subtype` must not be empty");
+    }
+
+    if (type.equals(MEDIA_TYPE_WILDCARD) && !subtype.equals(MEDIA_TYPE_WILDCARD)) {
+        throw MediaTypeException("Wildcard type is legal only in '*/*' (all types)");
+    }
+
+    this->type = type;
+    this->subtype = subtype;
+}
+
+MediaType::MediaType(const String &type, const String &subtype, const HashMap<String, String> &parameters) {
+
+    if (type.isEmpty()) {
+        throw MediaTypeException("`type` must not be empty");
+    }
+
+    if (subtype.isEmpty()) {
+        throw MediaTypeException("`subtype` must not be empty");
+    }
+
+    if (type.equals(MEDIA_TYPE_WILDCARD) && !subtype.equals(MEDIA_TYPE_WILDCARD)) {
+        throw MediaTypeException("Wildcard type is legal only in '*/*' (all types)");
+    }
+
+    this->type = type;
+    this->subtype = subtype;
+    this->parameters = parameters;
+}
+
+const HashMap<String, String> &MediaType::getParameters() const {
+    return parameters;
```

Comment:
**this->** helps read code easier and in sync with the use elsewhere
You can also check this: https://stackoverflow.com/questions/577243/is-there-any-reason-to-use-this